### PR TITLE
bpo-29878: Add global instances of int for 0 and 1.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
For internal use only.

### CODE REVIEW

**Review by mdickinson — APPROVED** [review]
LGTM

File: Objects/longobject.c [hunk 1]
```diff
@@ -2551,14 +2554,12 @@ long_divrem(PyLongObject *a, PyLongObject *b,
...
-        *pdiv = (PyLongObject*)PyLong_FromLong(0);
-        if (*pdiv == NULL)
-            return -1;
         *prem = (PyLongObject *)long_long((PyObject *)a);
         if (*prem == NULL) {
-            Py_CLEAR(*pdiv);
             return -1;
         }
+        *pdiv = (PyLongObject*)_PyLong_Zero;
+        Py_INCREF(_PyLong_Zero);
```
Comment (vstinner) [inline Objects/longobject.c]: Whenever possible, even if we are protected by the GIL, I prefer to have Py_INCREF before assignement, for consistency.

File: Objects/longobject.c [hunk 2]
```diff
@@ -5520,6 +5513,8 @@ PyLong_Fini(void)
     /* Integers are currently statically allocated. Py_DECREF is not
        needed, but Python must forget about the reference or multiple
        reinitializations will fail. */
+    Py_DECREF(_PyLong_One);
+    Py_DECREF(_PyLong_Zero);
```
Comment (vstinner) [inline Objects/longobject.c]: Py_CLEAR() would be safer if PyLong_Init() is called again (Py_Initialized called multiple times), no?

File: Objects/rangeobject.c [hunk 3]
```diff
@@ -83,17 +83,10 @@ range_new(PyTypeObject *type, PyObject *args, PyObject *kw)
...
-            return NULL;
-        }
-        step = PyLong_FromLong(1);
-        if (!step) {
-            Py_DECREF(stop);
-            Py_DECREF(start);
-            return NULL;
-        }
+        start = _PyLong_Zero;
+        Py_INCREF(start);
```
Comment (vstinner) [inline Objects/rangeobject.c]: ditto, in general I prefer to start with INCREF.

**Review by vstinner — COMMENTED** [review]
I'm really happy to see all these code go away! It's painful to have to check for errors on obvious things like that numbers 0 and 1!

The overall change LGTM, *but* IMHO it would be safer to use Py_CLEAR() in PyLong_Fini(). I have propose a minor coding style change.

**Review by vstinner — APPROVED** [review]

**Pull request merged**